### DISCUSSION TITLE
FireTV SlingerPlayer, cannot access Slinger Server remotely. Please help

### DISCUSSION THREAD

**Discussion by benson-burner1** [GerryDazoo/SlingerPlayer#34] [Q&A]
I have read every discussion, read me file here and elsewhere but cannot get my FireTV stick to access Slinger Server remotely.
I use this address in slingerplayer, (wan://external_ip_address:external_port). I have tried, (http://external_ip_address:external_port) as well. I have set my router to 
![VS](https://github.com/user-attachments/assets/b9945402-3a9c-492f-b111-dfb08a0f1bbc).![image](https://github.com/user-attachments/assets/434415d5-6959-4fd2-bcbb-a42c52b3a7c8)
I've tried using HTTP instead of FTP. I've even put the SlingerServer in the DMZ on router. What am I doing wrong? It works great over LAN. It will connect using the (http://external_ip_address:8080) but while on LAN. So I'm thinking my router is forwarding the address. I don't need it locally. I'm doing this for my mom who loves watching football (Chiefs) but with so many networks airing NFL, she gets very confused. I want her to just be able to go to the SlingerPlayer on FireTV and I will sling the game for her. But, I cannot get it to work remotely. Please help?

**Comment by GerryDazoo** [comment]
use http://external_ip:64530.
The WAN: option is only valid on the command line and will cause the player to auto-populate the list of available sling boxes on your local network. From your post it looks like you only have one box defined in your config.ini so this isn't needed,

**Reply by benson-burner1** [reply]
Thank you, Gerry, for responding so quickly. I will try it.
I'm thinking it's my external_ip address. It shows me something on my router interface, but when I use whatsmyip.com, it shows something completely different.
This is what I have adjusted in config.ini;
![config](https://github.com/user-attachments/assets/2a0e1bd9-5702-4a0a-a636-878dc90a0369)
Again, thank you.

**Comment by GerryDazoo** [comment]
Generally you would use the IP address that whatismyip reports but....
You may be the victim of your ISP "double NATting" you to minimize their costs for routable IP addresses. If your router says your external IP address is in one of these subnets (10.x.x.x, 192.168.x.x, 172.16-31.x.x) then you maybe out of luck unless you can get a fixed external IP address from your ISP usually at an additional cost per month.

**Reply by benson-burner1** [reply]
That was my thought. I live in an apartment that has free internet to everyone. I was thinking the apartment building gets the external ip and the units use a router to share that service. Hence, no need for a modem. I was thinking about trying the cloud base to stream to?

**Reply by benson-burner1** [reply]
Would a VPN or DynDNS solve my problem?

**Reply by GerryDazoo** [reply]
A VPN or DynDNS won't help if you're double NATed. Does your building have a IT guy. Maybe they would be willing to port forward 64530 on the "master" router to your box in your apt. Or ask them what it would take to get an external routable IP address. It doesn't have to be static because you can use a service like DynDNS to get packets to your apt if you've got a DHCP reservation.
A free cloud-based VM could be used to route the packets through a tunnel from your apt to the cloud and from the cloud to your player.

**Reply by benson-burner1** [reply]
Thank you so much for all your help and advise. I've been researching best ways to get around this. I did call the company that provides internet to these apartments. They do not or will not do anything for me. 
Thanks again.

**Reply by garabik** [reply]
You might want to check if your free internet provides (accessible from the outside) IPv6. It is a matter of a 2-line patch to modify slingbox_server to serve both ipv4 and ipv6, and it works at least with the Android SlingerPlayer out of the box.

If not, [n2n](https://github.com/ntop/n2n) might be a (cheap) way to tunnel your traffic, but requires endpoints running at both locations; in the days of working public miredo servers I would have advised to look into using miredo, but alas this is unavailable already for some years (you can run your own miredo server, but at that point you are probably better off installing your own OpenVPN).

**Reply by GerryDazoo** [reply]
I'm working on a new version of the server to fix some reported bugs and add a little functionality. Can you share your 2 line patch for IPV6 please.

Something like OpenVPN would work but the VPN connection would need to be make from where the server is to the player. So setting that up and keeping it alive from the server side so you can stream whenever you want is a little problematic. 
besides OPenVPN using SSH with a reverse tunnel would work as well all the hard work is the same, setting up the connection and keeping it alive. This is where a cheap/free could VM would come in handy. The connection would be from the server to the cloud and any player anywhere would be able to connect through it.

**Reply by garabik** [reply]
In linux, if the `net.ipv6.bindv6only` is set to 0 (usually the default), and with python >= 3.8, it is enough to change `socket.AF_INET` to `socket.AF_INET6`  when creating the ConnectionManagerSocket, so that the lines read `ConnectionManagerSocket = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)`. And it binds both to ipv4 and ipv6.

Since the default can be different and you can run it on a different OS (and then it might bind to ipv6 only), I would not suggest making it the default, but introduce a configuration parameter or a command line switch or something. There is also the `socket.has_dualstack_ipv6()` function that can be used to test if the system has dualstack support, so that might be the way to go (I have rather limited possibility to test it on Mac or Windows).
